lib/em-hiredis/redis_client.rb

```diff
@@ -313,15 +313,21 @@ def factory_connection
 
         connection.on(:connected) {
           maybe_auth(connection).callback {
-            maybe_select(connection).callback {
-              @command_queue.each { |command_df, command, args|
-                connection.send_command(command_df, command, args)
+            connection.ping.callback {
```

Comment:
Out of interest, do you happen to know whether you need to be authorised to send a `ping`? I would assume so but I don't know for sure.

Reply:
yes, you need to be authorised to send a command (`ping` in this case).

Reply:
I do. I tried locally to run redis with auth, and tried to send ping before `auth` – it was not possible. To send any command to redis you need to be authorised.